test/com/facebook/buck/android/MergeAndroidResourcesStepTest.java

```diff
@@ -425,6 +427,49 @@ public void testGenerateRDotJavaWithResourceUnionPackage() throws IOException {
     assertThat(res2java, StringContains.containsString("id2"));
   }
 
+  @Test
+  public void testGenerateRDotJavaWithRName() throws IOException {
+    BuildTarget res1Target = BuildTargetFactory.newInstance("//:res1");
+    RDotTxtEntryBuilder entriesBuilder = new RDotTxtEntryBuilder();
+    entriesBuilder.add(
+        new RDotTxtFile(
+            "com.res1",
+            BuildTargets.getGenPath(
+                entriesBuilder.getProjectFilesystem(),
+                res1Target,
+                "__%s_text_symbols__/R.txt").toString(),
+            ImmutableList.of("int id id1 0x7f020000")));
```

Comment:
I would suggest adding a second value and testing that both symbols appear in the output and are properly initialized to distinct ids.